gnovm/pkg/gnolang/type_check.go

```diff
@@ -279,6 +279,29 @@ func checkValDefineMismatch(n Node) {
 	panic(fmt.Sprintf("assignment mismatch: %d variable(s) but %d value(s)", numNames, numValues))
 }
 
+func checkExprIsAssignable(exp Expr) {
+	switch exp.(type) {
+	case *NameExpr,
+		*BasicLitExpr,
+		*BinaryExpr,
+		*CallExpr,
+		*IndexExpr,
+		*SelectorExpr,
+		*SliceExpr,
+		*StarExpr,
+		*RefExpr,
```

Comment:
consider this:
```go
package main

func main() {
	t := &struct{}
	t := *struct{}
}
```
These (but not limited to) should be checked recursively, right?

Reply:
did a bit refactoring
done here: https://github.com/gnolang/gno/pull/2896/commits/0646ab109fe22f6ed742fd791ff348bad18942ef